nautobot/extras/templates/extras/job.html

```diff
@@ -146,13 +146,32 @@
 
                     // Toggle schedule name and schedule time field rows.
                     ['#id__schedule_name', '#id__schedule_start_time'].forEach((selector) => {
-                        const row = document.querySelector(selector).parentElement.parentElement;
+                        const input = document.querySelector(selector);
+                        const row = input?.parentElement.parentElement;
                         row.classList.toggle('d-none', immediately);
+                        const label = row.querySelector('label');
+
+                        if (label) {
+                            label.classList.add('nb-required');
+                            if(!immediately) {
+                                input.setAttribute("required", immediately);
+                            }
+                        }
```

Comment:
what's missing here is some kind of opposite condition. right now, this code correctly sets `required` but it never removes it when fields become invisible (this is the case when I, for example, set "Once in the future" but change my mind and switch back to "Once immediately")

```suggestion
                        label?.classList.toggle('nb-required', !immediately);
                        if (immediately) {
                            input.removeAttribute('required');
                        } else {
                            input.setAttribute('required', '');
                        }
```

I also did some code lifting in this suggestion:
1. used optional chaining (`?` operator) instead of `if` condition to assert `label` existence - this is generally preferred syntax in JavaScript
2. flipped the other condition to comply with this eslint rule, which is enabled for JavaScript files: https://eslint.org/docs/latest/rules/no-negated-condition
3. set input attribute to `required` instead of `required="true"` - it doesn't matter much, but actually all that HTML needs to mark field as required is just the attribute, regardless of its value